Improving clang-format version parsing

On Ubuntu 20.04 I installed clang-format 10 and 11 but [Github action gives an error](https://github.com/BlueBrain/CoreNeuron/pull/456/checks?check_run_id=1714345359):

```
-- Found PythonInterp: /usr/bin/python3 (found suitable version "3.8.5", minimum required is "3.5") 
-- Found ClangFormat: /usr/bin/clang-format-11  
CMake Error at CMake/hpc-coding-conventions/cpp/CMakeLists.txt:48 (message):
 Unsupported Clang-Format version .  Supported versions are 7 or higher.
```

Checking version string under docker image I see:

```
root@7c1316599d1d:/# clang-format-11 --version
Ubuntu clang-format version 11.0.0-2~ubuntu20.04.1
root@7c1316599d1d:/# clang-format-10 --version
clang-format version 10.0.0-4ubuntu1
```

--- a/cpp/cmake/FindClangFormat.cmake
+++ b/cpp/cmake/FindClangFormat.cmake
@@ -34,7 +34,7 @@ else()
                NAMES clang-format
                      clang-format-11
                      clang-format-10
-                     clang-format-8
+                     clang-format-9
                      clang-format-8
                      clang-format-7
                      clang-format-6.0

--- a/cpp/cmake/FindClangFormat.cmake
+++ b/cpp/cmake/FindClangFormat.cmake
@@ -58,9 +58,8 @@ if(ClangFormat_EXECUTABLE)
                   ERROR_QUIET OUTPUT_STRIP_TRAILING_WHITESPACE)
 
   if(ClangFormat_version MATCHES ".*clang-format version .*")
-    # ClangFormat_version sample: "clang-format version 3.9.1-4ubuntu3~16.04.1""
+    # ClangFormat_version sample: "clang-format version 3.9.1-4ubuntu3~16.04.1"
     # ClangFormat_version sample: "Ubuntu clang-format version 11.0.0-2~ubuntu20.04.1"
-    # (tags/RELEASE_391/rc2)"
     string(REGEX
            REPLACE ".*clang-format version ([.0-9]+).*"
                    "\\1"

--- a/cpp/cmake/FindClangFormat.cmake
+++ b/cpp/cmake/FindClangFormat.cmake
@@ -32,7 +32,9 @@ if(ClangFormat_FIND_VERSION AND ClangFormat_FIND_VERSION_EXACT)
 else()
   find_program(ClangFormat_EXECUTABLE
                NAMES clang-format
-                     clang-format-9
+                     clang-format-11
+                     clang-format-10
+                     clang-format-8
                      clang-format-8
                      clang-format-7
                      clang-format-6.0
@@ -55,11 +57,12 @@ if(ClangFormat_EXECUTABLE)
                   OUTPUT_VARIABLE ClangFormat_version
                   ERROR_QUIET OUTPUT_STRIP_TRAILING_WHITESPACE)
 
-  if(ClangFormat_version MATCHES "^clang-format version .*")
-    # ClangFormat_version sample: "clang-format version 3.9.1-4ubuntu3~16.04.1
+  if(ClangFormat_version MATCHES ".*clang-format version .*")
+    # ClangFormat_version sample: "clang-format version 3.9.1-4ubuntu3~16.04.1""
+    # ClangFormat_version sample: "Ubuntu clang-format version 11.0.0-2~ubuntu20.04.1"
     # (tags/RELEASE_391/rc2)"
     string(REGEX
-           REPLACE "clang-format version ([.0-9]+).*"
+           REPLACE ".*clang-format version ([.0-9]+).*"
                    "\\1"
                    ClangFormat_VERSION
                    "${ClangFormat_version}")

